### PR TITLE
test/mqtt: Improve multi PDU parsing

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: 6592

## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: [6592](https://redmine.openinfosecfoundation.org/issues/6592)

### CODE REVIEW

**Review by catenacyber — CHANGES_REQUESTED** [review]
Should fail to show bug from redmine ticket, and so use a pcap that gets the tricky part

File: tests/mqtt-frames-xpdu/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Description
+===========
+Test MQTT frames[Pdu, Header, Data].
+
+PCAP
+====
+PCAP comes from the suricata verify test[mqtt5-pub-userpass]
```
Comment (catenacyber) [inline tests/mqtt-frames-xpdu/README.md:7]: Inconsistent with 
> pcap: ../mqtt5-unsub-userpass/input.pcap

File: tests/mqtt-frames-xpdu/test.rules [hunk 2]
```diff
@@ -0,0 +1,4 @@
+alert mqtt any any -> any any (msg:"mqtt frame: pdu 1"; frame:pdu; content: "|a2 0b 00 02|"; startswith; bsize:26; sid:1;)
```
Comment (catenacyber) [inline tests/mqtt-frames-xpdu/test.rules:1]: This should be `bsize: 13` cf Wireshark highlighting `a20b0002000006746f70696358`

**Comment by catenacyber** [comment]
I guess new PRs would be cleaner ;-)

**Review by victorjulien — APPROVED** [review]

File: tests/mqtt-frames-xpdu/test.rules [hunk 3]
```diff
@@ -0,0 +1,4 @@
+alert mqtt any any -> any any (msg:"mqtt frame: pdu 1"; frame:pdu; content: "|a2 0b 00 02|"; startswith; bsize:13; sid:1;)
+alert mqtt any any -> any any (msg:"mqtt frame: header"; frame:header; content: "|a2|"; startswith; bsize: 2; sid:2;)
```
Comment (catenacyber) [inline tests/mqtt-frames-xpdu/test.rules:2]: We could have the whole bytes of the frame, but fair enough for me

File: tests/mqtt-frames-xpdu/test.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
+pcap: ../mqtt5-unsub-userpass/input.pcap
+
+requires:
+  min-version: 8
+
+args:
+ - -k none
+ - --set outputs.1.eve-log.types.1.frame.enabled=yes
```
Comment (catenacyber) [inline tests/mqtt-frames-xpdu/test.yaml:8]: @jasonish is this robust enough ?

Alternative is to use own suricata.yaml, right ?
Or not have frames output, just alerts...

**Review by catenacyber — APPROVED** [review]
Good for me.

Just forwarded a question to Jason about `--set outputs.1.eve-log.types.1.frame.enabled=yes`

File: tests/mqtt-frames-xpdu/test.yaml (hunk 4 above)
Comment (jasonish) [reply tests/mqtt-frames-xpdu/test.yaml:8]: Yeah, but should we ever overhaul the default config it will break.. But so will many many tests.

**Pull request closed without merging**